src/fileformat/file_format/pe/pe_format.cpp

```diff
@@ -3350,5 +3409,320 @@ const VisualBasicInfo* PeFormat::getVisualBasicInfo() const
+		}
+
+		// scan section marked uninitialized but contains data
+		if ((flags & PELIB_IMAGE_SCN_CNT_UNINITIALIZED_DATA) &&
+			(sec->getOffset() != 0 || sec->getSizeInFile() != 0))
+		{
+			anomalies.emplace_back(std::make_pair<std::string, std::string>("uninitSecHasData", 
+				"Section " + msgName + " is marked uninitialized but contains data"));
+		}
+
+		// scan sizeOfRawData of section is 0
+		if (sec->getSizeInFile() == 0)
+		{
+			anomalies.emplace_back(std::make_pair<std::string, std::string>("noRawDataSec", 
+				"Section " + msgName + " has zero SizeOfRawData"));
+		}
+
+		for (std::size_t j = i + 1; j < nSecs; j++)
+		{
+			auto cmpSec = getSection(j);
+			if (!cmpSec)
+			{
+				continue;
+			}
+
+			// scan for duplicit section names
+			const auto cmpName = cmpSec->getName();
+			if (!name.empty() && name == cmpName)
+			{
+				if (std::find(duplSections.begin(), duplSections.end(), name) == duplSections.end())
+				{
+					anomalies.emplace_back(std::make_pair<std::string, std::string>("duplSecNames",
+						"Multiple sections with name " + name));
+					duplSections.push_back(name);
+				}
+			}
+
+			// scan for overlapping sections
+			auto secStart = sec->getOffset();
+			auto secEnd = secStart + sec->getSizeInFile();
+			auto cmpSecStart = cmpSec->getOffset();
+			auto cmpSecEnd = cmpSecStart + cmpSec->getSizeInFile();
+			if ((secStart <= cmpSecStart && cmpSecStart < secEnd) ||
+				(cmpSecStart <= secStart && secStart < cmpSecEnd))
+			{
+				const std::string cmpMsgName = (cmpName.empty()) ? numToStr(cmpSec->getIndex()) : cmpName;
+				anomalies.emplace_back(std::make_pair<std::string, std::string>("overlappingSecs",
+					"Sections " + msgName + " and " + cmpMsgName + " overlap"));
+			}
+		}
+	}
+}
+
+/**
+ * Scan for section anomalies
+ */
+void PeFormat::scanForResourceAnomalies()
+{
+	if (!resourceTable)
+	{
+		return;
+	}
+
+	for (std::size_t i = 0; i < resourceTable->getNumberOfResources(); i++)
+	{
+		auto res = resourceTable->getResource(i);
+		if (!res)
+		{
+			continue;
+		}
+
+		std::size_t nameId;
+		std::string msgName = (res->getNameId(nameId)) ? numToStr(nameId) : "<unknown>";
+
+		// scan for resource size over 100MB
+		if (res->getSizeInFile() >= 100000000UL)
+		{
+			anomalies.emplace_back(std::make_pair<std::string, std::string>("largeRes",
+				"Resource " + msgName + " has size over 100MB"));
+		}
+
+		// scan for resource stretched over multiple sections
+		unsigned long long resAddr;
+		if (getAddressFromOffset(resAddr, res->getOffset()) &&
+			isObjectStretchedOverSections(resAddr, res->getSizeInFile()))
+		{
+			anomalies.emplace_back(std::make_pair<std::string, std::string>("stretchedRes",
+				"Resource " + msgName + " is stretched over multiple sections"));
+		}
+	}
+}
+
+/**
+ * Scan for import anomalies
+ */
+void PeFormat::scanForImportAnomalies()
+{
+	// scan for import stretched over multiple sections
+	for(auto&& impRange : formatParser->getImportDirectoryOccupiedAddresses())
```

Comment:
Do we really need a forwarding reference in this case? Wouldn't `const auto&` suffice?
```cpp
for(const auto& impRange : formatParser->getImportDirectoryOccupiedAddresses())
```
As Howard Hinnant puts it, liberate use of `auto&&` results in so-called [confuscated code](http://stackoverflow.com/a/13130795/2580955): code that unnecessarily confuses people.

Similar piece of code appears in `PeFormat::scanForExportAnomalies()`.